project-base/storefront/components/PaymentsInOrderSelect/PaymentsInOrderSelect.tsx

```diff
@@ -0,0 +1,123 @@
+
+type PaymentsInOrderSelectProps = {
+    orderUuid: string;
+    orderPaymentType: string | undefined;
+    canPaymentBeRepeated: boolean;
+};
+
+export const PaymentsInOrderSelect: FC<PaymentsInOrderSelectProps> = ({
+    orderUuid,
+    canPaymentBeRepeated,
+    orderPaymentType,
+}) => {
+    const { t } = useTranslation();
+    const { changePaymentInOrderFetching, changePaymentInOrderHandler } = useChangePaymentInOrder();
+    const [selectedPaymentUuid, setSelectedPaymentUuid] = useState<string | undefined>();
+
+    const [{ data: orderAvailablePaymentsData, fetching: orderAvailablePaymentsFetching }] =
+        useOrderAvailablePaymentsQueryApi({
+            variables: { orderUuid },
+        });
+
+    const currentOrderPayment = orderAvailablePaymentsData?.orderPayments.currentPayment;
+
+    useEffect(() => {
+        setSelectedPaymentUuid(currentOrderPayment?.uuid);
+    }, [currentOrderPayment?.uuid]);
+
+    if (orderAvailablePaymentsFetching) {
+        return <SpinnerIcon className="mx-auto mt-4 block w-12" />;
+    }
+
+    if (!orderAvailablePaymentsData) {
+        return null;
+    }
+
+    return (
+        <div className="mt-6 flex flex-col items-center gap-6">
+            {orderPaymentType === PaymentTypeEnum.GoPay && canPaymentBeRepeated && currentOrderPayment && (
+                <div className="flex w-full flex-col items-center">
+                    <TransportAndPaymentListItem
+                        key={currentOrderPayment.uuid}
+                        className="order-none"
+                        isActive={selectedPaymentUuid === currentOrderPayment.uuid}
+                    >
+                        <Radiobutton
+                            checked={currentOrderPayment.uuid === selectedPaymentUuid}
+                            id={currentOrderPayment.uuid}
+                            name="payment"
+                            value={currentOrderPayment.uuid}
+                            label={
+                                <TransportAndPaymentSelectItemLabel
+                                    description={currentOrderPayment.description}
+                                    image={currentOrderPayment.mainImage}
+                                    name={currentOrderPayment.name}
+                                    price={currentOrderPayment.price}
+                                />
+                            }
+                            onChangeCallback={() => setSelectedPaymentUuid(currentOrderPayment.uuid)}
+                        />
+                    </TransportAndPaymentListItem>
+                    <GoPayGateway
+                        requiresAction
+                        className="mt-5"
+                        initialButtonText={t('Repeat payment')}
+                        isDisabled={selectedPaymentUuid !== currentOrderPayment.uuid}
+                        orderUuid={orderUuid}
+                    />
+                </div>
+            )}
```

Comment:
I am trying to understand this code. You use `currentOrderPayment` here. Is it sure that this is going to be go pay? Is it because if `orderPaymentType` is go pay, and we have `currentOrderPayment`, we know it is going to be go pay? I am not sure if I quite follow.

Reply:
Good point, I added a condition where I check if currentOrderPayment is gopay